File: src/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {

    /**
     * @param data
     * @param reverse
     */
    private static void bubbleSort(int[] data, boolean reverse) {

        if (data.length == 1) {
            return;
        }

        for (int i = 0; i < data.length - 1; i++) {

            int tmp = 0;

            for (int j = 0; j < data.length - i - 1; j++) {
                if (reverse) {
                    if (data[j] >= data[j + 1]) {
                        tmp = data[j];
                        data[j] = data[j + 1];
                        data[j + 1] = tmp;
                    }
                } else {
                    if (data[j] <= data[j + 1]) {
                        tmp = data[j + 1];
                        data[j + 1] = data[j];
                        data[j] = tmp;
                    }

                }
            }

        }
    }


    public static void main(String[] args){
        int[] input = {2,41,25,562,6,25,27,56,67,34,52,25,7,54,54,123,14};
        bubbleSort(input, false);
        System.out.println(Arrays.toString(input));

    }


}

File: BubbleSort.py
# coding:utf-8


def bubble_sort(data, reverse):
    """

    :param data: list
    :param reverse: boolean -> true: small -> big, vice versa.
    :return: sorted list
    """
    if len(data) == 1:
        return
    for i in range(len(data) - 1):
        for j in range(len(data) - i - 1):
            if reverse:
                if data[j] >= data[j + 1]:
                    tmp = data[j]
                    data[j] = data[j + 1]
                    data[j + 1] = tmp
            else:
                if data[j] <= data[j + 1]:
                    tmp = data[j]
                    data[j + 1] = data[j]
                    data[j] = tmp


if __name__ == '__main__':
    a = [19, 25, 2, 1, 4, 23, 5, 7, 67, 3, 3, 1, 8, 74]
    bubble_sort(a, True)
    print(a)
